setup.sh

```diff
@@ -3,14 +3,53 @@ GREEN='\033[0;32m'
 CYAN='\033[0;36m'
 NC='\033[0m' # No Color
 
-printf "${CYAN}Get UI Bakery license - ${GET_KEY_LINK}:\n${NC}"
-printf '\n'
-printf "Enter license key:\n"
-while read license; do
-  test "$license" != "" && break
-  printf "${RED}License key is required!${NC}\n"
-  printf "Enter license key:"
+printf "${CYAN}Starting license setup...\n${NC}"
+echo "Do you already have a UI Bakery license key?"
+echo "Y - enter your license key, N - start a free trial."
+while read have_license_key_y_n; do
+  if [[ "$have_license_key_y_n" == "Y" ]] || [[ "$have_license_key_y_n" == "y" ]] || [[ "$have_license_key_y_n" == "" ]]; then
+    CUSTOM_LICENSE_KEY="YES"
+    break
+  elif [[ "$have_license_key_y_n" == "N" ]] || [[ "$have_license_key_y_n" == "n" ]]; then
+    echo "Enter working email address to generate the free trial license of UI Bakery:"
+    while read email; do
+
+      email_regex="^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$"
+      if [[ $email =~ $email_regex ]] ; then
+        break
+      else
+        printf "${RED}Valid email is required to request a license${NC}\n"
+        printf "Enter email:"
+      fi
+    done
+
+    printf "EMAIL: ${email}\n\n"
+    CUSTOM_LICENSE_KEY="NO"
+    break
+  else
+   echo "Y - yes, I have have a license key; N - no, I want to start a free trial"
```

Comment:
looks like ought to be the same as line 8

Reply:
fixed